Source/qol/autopickup.cpp

```diff
@@ -57,4 +74,41 @@ void AutoGoldPickup(int pnum)
 	}
 }
 
+void AutoItemPickup(int pnum)
+{
+	for (auto pathDir : PathDirs) {
+		Point tile = Players[pnum].position.tile + pathDir;
+		if (dItem[tile.x][tile.y] != 0) {
+			int itemIndex = dItem[tile.x][tile.y] - 1;
+			auto &item = Items[itemIndex];
+			if (AutoPlaceItemInInventory(Players[pnum], item, false) || AutoPlaceItemInBelt(Players[pnum], item, false)) {
+				if (sgOptions.Gameplay.nHealPotionPickup > numMiscItemsInInv(IMISC_HEAL) && item._iMiscId == IMISC_HEAL) {

```

Comment:
I don't think it'll make much of a difference in release builds but it's nice to have the simple condition first:
```suggestion
				if (item._iMiscId == IMISC_HEAL && sgOptions.Gameplay.nHealPotionPickup > numMiscItemsInInv(IMISC_HEAL)) {
```

lets the compiler short circuit and avoid running the loop in `numMiscItemsInInv` if it's not smart enough to recognise that optimisation.

Reply:
I trust my compiler in this case ;)

Reply:
Turns out the built in local AND/OR operators enforce sequence ordering. We don't override operator&&(bool, bool) so this will always run the loop before checking the item is actually the correct type.

> Every value computation and side effect of the first (left) argument of the built-in logical AND operator && and the built-in logical OR operator || is sequenced before every value computation and side effect of the second (right) argument.
https://en.cppreference.com/w/cpp/language/eval_order

Reply:
Seriously.... OK I'll swap it around.